providers/evercrypt/fst/EverCrypt.Hash.fst

```diff
@@ -170,7 +268,13 @@ let init #a s =
   | SHA2_384_s p -> Hacl.Hash.SHA2.init_384 p
   | SHA2_512_s p -> Hacl.Hash.SHA2.init_512 p
   | Blake2S_s p -> let _ = Hacl.Hash.Blake2.init_blake2s_32 p in ()
+  | Blake2S_128_s _ p ->
+      if EverCrypt.TargetConfig.hacl_can_compile_vec128 then
```

Comment:
Why such checks (same question for other functions like `update2`)? There is no way we created a `Blake2S_128_s` state if we don't have the proper flag. And I guess if we still managed to do so with some hacking, isn't it better to fail by trying to use illegal instructions than having the functions silently behave as no-ops?

Reply:
this is to prevent a link-time error; without this check, EverCrypt_Hash.o contains a reference to the symbol Blake2s_128_init; if you are on a platform for which vec128 is not available, then Blake2s_128_init is not compiled and you get a link-time error

with this check, there is an #ifdef surrounding the reference to Blake2s_128_init and all is well

Reply:
Indeed, I had forgotten that those checks were extracted to #ifdef